src/main/java/org/cactoos/iterable/Sticky.java

```diff
@@ -51,14 +52,19 @@ public Sticky(final X... src) {
      */
     public Sticky(final Iterable<X> iterable) {
         super(
-            new org.cactoos.scalar.Sticky<>(
-                () -> {
-                    final Collection<X> temp = new LinkedList<>();
-                    for (final X item : iterable) {
-                        temp.add(item);
-                    }
-                    return temp;
-                }
+            new IterableOf<>(
+                new Mapped<>(
+                    Iterable::iterator,
+                    new org.cactoos.scalar.Sticky<Iterable<X>>(
+                        () -> {
+                            final Collection<X> temp = new LinkedList<>();
+                            for (final X item : iterable) {
```

Comment:
@victornoel is it possible to refactor this part and use an object-oriented approach?

Reply:
@scristalli I believe it's out of scope of the issue solved, you can open an issue in cactoos if you want. I tried to think about it, but it's not clear what is best between using `ListOf` + `scalar.Sticky`, or directly `list.Sticky`, etc. You can add this information in the issue you create for example.